RLTest/redis_std.py

```diff
@@ -349,12 +351,29 @@ def _stopProcess(self, role):
                         p.wait()
                     except:
                         pass
-            process.terminate()
-            while True:
-                if process.poll() is None:  # None returns if the processes is not finished yet, retry until redis exits
-                    time.sleep(0.1)
-                else:
-                    break
+
+            if self.terminateRetries is None:
+                # ask once, then wait for process to exit
+                process.terminate()
+                while True:
+                    if process.poll() is None:  # None returns if the processes is not finished yet, retry until redis exits
+                        time.sleep(0.1)
+                    else:
+                        break
+            else:
+                # keep asking every few seconds until process has exited, otherwise kill
+                if self.terminateRetrySecs is None:
+                    self.terminateRetrySecs = 1
+                done = False
+                for i in range(1, self.terminateRetries):
+                    process.terminate()
+                    if process.poll() is None:  # None returns if the processes is not finished yet, retry until redis exits
+                        time.sleep(self.terminateRetrySecs)
+                    else:
+                        done = True
+                        break
+                if not done:
+                    process.kill()
```

Comment:
Shouldn't this fail the test?

Reply:
I donno. WDYT?
We have `require_clean_exit` and `require_clean_exit` in `RLTest` class, but not in `Env`. But maybe it'll proliferate.

Reply:
Seems to me we need to consider `require_clean_exit` if we kill the process, in order to be able to fail just as on any other unclean exit, e.g., as in PR #177 on failed flush, etc.